pkg/services/auth/jwt/key_sets.go

```diff
@@ -141,19 +146,35 @@ func (s *AuthService) initKeySet() error {
 			return err
 		}
 
-		s.keySet = keySetJWKS{jwks}
+		s.keySet = &keySetJWKS{jwks}
 	} else if urlStr := s.Cfg.JWTAuthJWKSetURL; urlStr != "" {
 		urlParsed, err := url.Parse(urlStr)
 		if err != nil {
 			return err
 		}
-		if urlParsed.Scheme != "https" {
+		if urlParsed.Scheme != "https" && setting.Env != setting.Dev {
 			return ErrJWTSetURLMustHaveHTTPSScheme
 		}
 		s.keySet = &keySetHTTP{
-			url:             urlStr,
-			log:             s.log,
-			client:          &http.Client{},
+			url: urlStr,
+			log: s.log,
+			client: &http.Client{
```

Comment:
question: just curious about why you had to specify these? Which of the default values had to be changed?

Reply:
I did not need to specify, but it's best practice to configure timeouts and the transport (ProxyFromEnv). I got the defaults from some of our other services, actually missing an exported "NewHTTPClient(timeout duration) *http.Client"  somewhere

Reply:
So it's mostly to be explicit about the timeouts etc? That makes sense, thanks for the explanation!